### PR TITLE
[benchmarks] speed up CI

### PR DESCRIPTION
**Pull request by pepeiborra**
* Disable parallel GC as it leads to occasional segfaults when combined with heap profiling
* Lower the frequency of heap profiles to speed up benchmarks (I'll make it configurable in a follow-up change)
* Reduce samples from 100 to 50 to speed up benchmarks

While I was there I also changed some of the artefact file extensions for consistency

### CODE REVIEW

File: shake-bench/src/Development/Benchmark/Rules.hs [hunk 1]
```diff
@@ -250,22 +251,16 @@ benchRules build benchResource MkBenchRules{..} = do
...
         let header = head csvContents
             results = tail csvContents
             header' = header <> ", maxResidency, allocatedBytes"
         results' <- forM results $ \row -> do
-            -- assume that the gcStats file can be guessed from the row id
-            -- assume that the row id is the first column
-            let id = takeWhile (/= ',') row
-            let gcStatsPath = dropFileName outcsv </> escapeSpaces id <.> "benchmark-gcStats"
-            (maxResidency, allocations) <- liftIO $
-                ifM (IO.doesFileExist gcStatsPath)
```
Comment (wz1000) [inline shake-bench/src/Development/Benchmark/Rules.hs:266]: Why remove this check?

**Review by jneira — APPROVED** [review]
LGTM, thans for improve ci times

File: shake-bench/src/Development/Benchmark/Rules.hs (hunk 1 above)
Comment (pepeiborra) [reply shake-bench/src/Development/Benchmark/Rules.hs:266]: In a previous change, I moved the logic that generates this file from ghcide-bench to shake-bench, so there is no more guessing involved

**Comment by pepeiborra** [comment]
I thought that the benchmark job was the bottleneck, but I've sent 3 PRs today and they are all still in the CircleCI queue after >7h ....

**Comment by konn** [comment]
Seems mergify got stuck, as condition requires benchfor 8.8.4 and 8.10.2.

**Pull request merged**